manager/integration/tests/test_node.py

```diff
@@ -0,0 +1,375 @@
+import time
+import common
+import pytest
+import os
+import subprocess
+
+from common import clients  # NOQA
+from common import Gi, SIZE
+from common import get_self_host_id
+
+SMALL_DISK_SIZE = (1 * 1024 * 1024)
+DEFAULT_DISK_PATH = '/var/lib/rancher/longhorn/'
+TEST_WRITE_SIZE = 1 * Gi
+TEST_FILE = '/var/lib/rancher/longhorn/test'
+WAIT_FOR_NODE_STATUS = 60
+
+
+def create_disk(client, vol_name, size, node_id):
+    # create a single replica volume and attach it to node
+    volume = create_volume(client, vol_name, size, node_id, 1)
+
+    for replica in volume["replicas"]:
+        id = replica["hostId"]
+        assert id != ""
+        assert replica["running"]
+
+    # prepare the disk in the host filesystem
+    disk_path = common.prepare_host_disk(volume["endpoint"], volume["name"])
+    return disk_path
+
+
+def clean_disk(client, *args):
+    # clean disk
+    for vol_name in args:
+        # umount disk
+        common.umount_host_disk(vol_name)
+        # clean volume
+        clean_volume(client, vol_name)
+
+    volumes = client.list_volume()
+    assert len(volumes) == 0
+    # clean all disks
+    common.cleanup_host_disk_directory()
+
+
+def get_update_disks(disks):
+    update_disk = []
+    for key, disk in disks.iteritems():
+        update_disk.append(disk)
+    return update_disk
+
+
+def test_node(clients):  # NOQA
+    # get a random client
+    for host_id, client in clients.iteritems():
+        break
+
+    # test node update
+    nodes = client.list_node()
+    assert len(nodes) > 0
+    lht_hostId = get_self_host_id()
+    node = client.by_id_node(lht_hostId)
+    node = client.update(node, allowScheduling=False)
+    assert not node["allowScheduling"]
+    node = client.by_id_node(lht_hostId)
+    assert not node["allowScheduling"]
+
+    node = client.update(node, allowScheduling=True)
+    assert node["allowScheduling"]
+    node = client.by_id_node(lht_hostId)
+    assert node["allowScheduling"]
+
+    disks = node["disks"]
+
+    # test add disk exception
+    with pytest.raises(Exception):
+        # add same disk by different mount path
+        disk = {"path": "/var/lib", "allowScheduling": True,
+                "storageMaximum": 5*Gi, "storageReserved": 2*Gi}
+        update_disk = get_update_disks(disks)
+        update_disk.append(disk)
+        node = node.diskUpdate(disks=update_disk)
+
```

Comment:
You can check the for error message if an exception is expected. See https://github.com/rancher/longhorn-tests/blob/master/manager/integration/tests/test_engine_upgrade.py#L126

Also, only the line will cause exception in the `with`, otherwise if any other code caused exception it will result in a false negative result.